app/controllers/users/passwords_controller.rb

```diff
@@ -75,36 +67,26 @@ def handle_successful_password_reset
       EmailNotifier.new(resource).send_password_changed_email
     end
 
-    def handle_unsuccessful_password_reset
-      if resource.errors[:reset_password_token].present?
-        handle_expired_token(resource)
+    def handle_unsuccessful_password_reset(result)
+      if result[:errors].include?('token_expired')
+        flash[:error] = t('devise.passwords.token_expired')
         redirect_to new_user_password_path
         return
       end
 
-      analytics.track_event(Analytics::PASSWORD_RESET_INVALID_PASSWORD, user_id: resource.uuid)
       render :edit
     end
 
     def mark_profile_inactive
       active_profile = resource.active_profile
-      return unless active_profile.present?
-      active_profile.update!(active: false)
-      analytics.track_event(
```

Comment:
Correct. LOA3 users who reset their passwords must then rely on a recovery code or lose their LOA3 status. Tracking that use case (LOA3 users who reset their passwords) is thus an important usability metric.